runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -1019,11 +1019,16 @@ J9::CodeGenerator::lowerTreeIfNeeded(
 
    if (node->getOpCode().isCall() &&
        !node->getSymbol()->castToMethodSymbol()->isHelper() &&
-       (node->getSymbol()->castToMethodSymbol()->getRecognizedMethod() == TR::java_lang_invoke_MethodHandle_invokeBasic))
+       ((node->getSymbol()->castToMethodSymbol()->getRecognizedMethod() == TR::java_lang_invoke_MethodHandle_invokeBasic) ||
+        (node->getSymbol()->castToMethodSymbol()->getRecognizedMethod() == TR::java_lang_invoke_MethodHandle_linkToStatic) ||
+        (node->getSymbol()->castToMethodSymbol()->getRecognizedMethod() == TR::java_lang_invoke_MethodHandle_linkToSpecial) ||
+        (node->getSymbol()->castToMethodSymbol()->getRecognizedMethod() == TR::java_lang_invoke_MethodHandle_linkToVirtual) ||
+        (node->getSymbol()->castToMethodSymbol()->getRecognizedMethod() == TR::java_lang_invoke_MethodHandle_linkToInterface)))
       {
-      // invokeBasic is signature-polymorphic, so the location of the method handle receiver object on the stack will be
-      // the number of arg slots in the invokeBasic call minus 1 accounting for the receiver slot. This value is is stored
-      // in vmThread.tempSlot so that the interpreter can locate the receiver object on the stack.
+      // invokeBasic and linkTo* are signature-polymorphic, so the location of the method handle
```

Comment:
The code was to pass location of the MethodHandle receiver for `invokeBasic`,  but `linkTo*` methods dont' have MethodHandle receiver. Maybe we should pass the number of argument slots (including receiver if exists), and VM can use this info differently for different methods. @babsingh If we were to write the number of argument slots to `tempSlot`, I believe VM has to modify your code in handling `invokeBasic` called from the JIT

Reply:
Receiver is the first arg. The receiver's location can be derived knowing the number of argument slots. @fengxue-IS to confirm?

Reply:
@nbhuiyan Can you change the code to write the number of argument slots to `tempSlot` instead of the location of the first argument? @babsingh VM has to modify the place that currently uses `tempSlot` 
https://github.com/eclipse/openj9/blob/3dcfa980467d67ac059129f7349fff6159469737/runtime/vm/BytecodeInterpreter.hpp#L8251